### PR TITLE
Cleanup Wasm tracing flags

### PR DESCRIPTION
**Pull request by Cellule**
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/3794)

### CODE REVIEW

File: lib/WasmReader/WasmBinaryReader.cpp [hunk 1]
```diff
@@ -214,6 +215,17 @@ SectionHeader WasmBinaryReader::ReadSectionHeader()
...
+    if (m_readerState == READER_STATE_FUNCTION)
+    {
+        return m_funcState.body;
+    }
+    return nullptr;
+}
+#endif
+
 #if DBG_DUMP
 void WasmBinaryReader::PrintOps()
```
Comment (Cellule) [inline lib/WasmReader/WasmBinaryReader.cpp]: I wasn't sure if we wanted to keep PrintOps.
In the current code base in doesn't do anything, because we print before actually parsing any opcode.

File: lib/WasmReader/WasmBinaryReader.cpp [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #include "WasmLimits.h"
 #if ENABLE_DEBUG_CONFIG_OPTIONS
 #include "Codex/Utf8Helper.h"
+#include <inttypes.h>
```
Comment (MikeHolman) [inline lib/WasmReader/WasmBinaryReader.cpp]: Why is this needed?

File: lib/WasmReader/WasmByteCodeGenerator.h [hunk 3]
```diff
@@ -159,6 +159,7 @@ namespace Wasm
         Js::WebAssemblyModule* GenerateModule();
         void GenerateFunctionHeader(uint32 index);
     private:
+        Js::FunctionBody* GetFunctionBody() const { return nullptr; }
```
Comment (MikeHolman) [inline lib/WasmReader/WasmByteCodeGenerator.h]: this seems bad

File: lib/WasmReader/WasmReader.h [hunk 4]
```diff
@@ -20,23 +20,19 @@
     }
 
 // Level of tracing
-#define DO_WASM_TRACE_ALL       PHASE_TRACE1(Js::WasmPhase)
-#define DO_WASM_TRACE_BYTECODE  DO_WASM_TRACE_ALL || PHASE_TRACE(Js::WasmBytecodePhase, GetFunctionBody())
-#define DO_WASM_TRACE_DECODER   DO_WASM_TRACE_ALL || PHASE_TRACE1(Js::WasmReaderPhase)
-  #define DO_WASM_TRACE_SECTION DO_WASM_TRACE_DECODER || PHASE_TRACE1(Js::WasmSectionPhase)
-#define DO_WASM_TRACE_LEB128    PHASE_TRACE1(Js::WasmLEB128Phase)
+#define WASM_TRACE_BODY_CHECK(phase) (GetFunctionBody() ? PHASE_TRACE(phase, GetFunctionBody()) : PHASE_TRACE1(phase))
```
Comment (MikeHolman) [inline lib/WasmReader/WasmReader.h]: I guess this is why you added the GetFunctionBody method, but without this context it is confusing

File: lib/WasmReader/WasmByteCodeGenerator.h (hunk 3 above)
Comment (Cellule) [reply lib/WasmReader/WasmByteCodeGenerator.h]: Agreed, I think there is only one case that needs it for the macro, so I'll just rewrite that part accordingly
